x/tss/keeper/grpc_query.go

```
@@ -58,3 +62,71 @@ func (q Querier) AssignableKey(c context.Context, req *types.AssignableKeyReques
+	ctx := sdk.UnwrapSDKContext(c)
+
+	valAddress, err := sdk.ValAddressFromBech32(req.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	validator := q.staker.Validator(ctx, valAddress)
+	if validator == nil {
+		return nil, fmt.Errorf("not a validator")
+	}
+
+	chains := q.nexus.GetChains(ctx)
+	keys := make(map[string]*types.ValidatorKeyResponse_Keys, 10)
+
+	keyRoles := []exported.KeyRole{exported.MasterKey, exported.SecondaryKey}
+
+	for _, chain := range chains {
+		for _, keyRole := range keyRoles {
+			if currentKey, found := q.keeper.GetCurrentKey(ctx, chain, keyRole); found {
+				if valKeys, ok := q.getSerializedMultisigKeys(ctx, currentKey, valAddress); ok {
+					keys[string(currentKey.ID)] = valKeys
+				}
+			}
+
+			if nextKey, found := q.keeper.GetNextKey(ctx, chain, keyRole); found {
+				if valKeys, ok := q.getSerializedMultisigKeys(ctx, nextKey, valAddress); ok {
+					keys[string(nextKey.ID)] = valKeys
+				}
+			}
+
+			oldActiveKeys, err := q.keeper.GetOldActiveKeys(ctx, chain, keyRole)
+			if err != nil {
+				return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, err.Error())
+			}
+
+			for _, oldActiveKey := range oldActiveKeys {
+				if valKeys, ok := q.getSerializedMultisigKeys(ctx, oldActiveKey, valAddress); ok {
+					keys[string(oldActiveKey.ID)] = valKeys
+				}
+			}
+		}
+	}
+
+	return &types.ValidatorKeyResponse{Keys: keys}, nil
+}
+
+func (q Querier) getSerializedMultisigKeys(ctx sdk.Context, key exported.Key, valAddress sdk.ValAddress) (*types.ValidatorKeyResponse_Keys, bool) {
+	if key.Type != exported.Multisig {
+		return nil, false
+	}
+
+	if pubkeys, ok := q.keeper.GetMultisigPubKeysByValidator(ctx, key.ID, valAddress); ok {
+		valKeys := make([][]byte, len(pubkeys))
+
+		for i, pubkey := range pubkeys {
+			wrappedKey := btcec.PublicKey(pubkey)
+			valKeys[i] = (&wrappedKey).SerializeCompressed()
+		}
+
+		return &types.ValidatorKeyResponse_Keys{Keys: valKeys}, true
+	}
+
+	return nil, false
```

```suggestion
	pubkeys, ok := q.keeper.GetMultisigPubKeysByValidator(ctx, key.ID, valAddress)
	if !ok {
	        return nil, false
	}
		
	valKeys := make([][]byte, len(pubkeys))

	for i, pubkey := range pubkeys {
		wrappedKey := btcec.PublicKey(pubkey)
		valKeys[i] = (&wrappedKey).SerializeCompressed()
	}

	return &types.ValidatorKeyResponse_Keys{Keys: valKeys}, true
```

---

always leave the success case for the end of a function and fail early.